src/configLoader.js

```diff
@@ -53,3 +45,20 @@ export function resolvePath(configPath) {
 
     return path.resolve(process.cwd(), configPath);
 }
+
+/*
+    With the introduction of environment variable subsitution, we need to restore environment
+    variables in the written config for consistency + not writing secrets to disk. This replacement
+    function assumes that Kong will never change the value of a config property that contains a
+    secret. We can currently assume that Kong will never change the value of any property, period.
+    This function is safe against Kong removing fields.
+*/
+ export function sanitizeConfigForSafeWrite(config, envPointers) {
+    for (let pointer in envPointers) {
+        if (jsonPointer.has(config, pointer)) {
+            jsonPointer.set(config, pointer, envPointers[pointer]);
+        }
```

Comment:
Do we want to log if the config somehow doesn't have the pointer? Seems unlikely to matter but if we are already doing the check adding the log line might save some headache in the future.

Reply:
i think that is absolutely a good idea, yeah